app/scripts/lib/account-tracker.test.js

```diff
@@ -53,103 +65,344 @@ describe('Account Tracker', () => {
-      accountTracker.store.updateState({ accounts: { ...mockAccounts } });
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
 
-      await accountTracker._updateAccount(VALID_ADDRESS);
+  describe('start', () => {
+    it('restarts the subscription to the block tracker and update accounts', async () => {
+      jest.spyOn(blockTrackerStub, 'addListener').mockImplementation();
+      jest.spyOn(blockTrackerStub, 'removeListener').mockImplementation();
+      const updateAccountsWithNetworkClientIdSpy = jest
+        .spyOn(accountTracker, '_updateAccounts')
+        .mockResolvedValue();
 
-      const newAccounts = accountTracker.store.getState();
+      accountTracker.start();
 
-      const expectedAccounts = {
-        accounts: {
-          [VALID_ADDRESS]: { address: VALID_ADDRESS, balance: UPDATE_BALANCE },
-          [VALID_ADDRESS_TWO]: {
-            address: VALID_ADDRESS_TWO,
-            balance: INITIAL_BALANCE_2,
-          },
-        },
-        currentBlockGasLimit: '',
-      };
+      expect(blockTrackerStub.removeListener).toHaveBeenNthCalledWith(
+        1,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(blockTrackerStub.addListener).toHaveBeenNthCalledWith(
+        1,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(updateAccountsWithNetworkClientIdSpy).toHaveBeenNthCalledWith(1); // called first time with no args
 
-      expect(newAccounts).toStrictEqual(expectedAccounts);
+      accountTracker.start();
+
+      expect(blockTrackerStub.removeListener).toHaveBeenNthCalledWith(
+        2,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(blockTrackerStub.addListener).toHaveBeenNthCalledWith(
+        2,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+      expect(updateAccountsWithNetworkClientIdSpy).toHaveBeenNthCalledWith(2); // called second time with no args
+
+      accountTracker.stop();
     });
+  });
 
-    it('should not change accounts if the passed address is not in accounts', async () => {
-      accountTracker.store.updateState({ accounts: { ...mockAccounts } });
+  describe('stop', () => {
+    it('ends the subscription to the block tracker', async () => {
+      jest.spyOn(blockTrackerStub, 'removeListener').mockImplementation();
 
-      await accountTracker._updateAccount('fake address');
+      accountTracker.stop();
 
-      const newAccounts = accountTracker.store.getState();
+      expect(blockTrackerStub.removeListener).toHaveBeenNthCalledWith(
+        1,
+        'latest',
+        accountTracker._updateForBlock,
+      );
+    });
+  });
+
+  describe('startPollingByNetworkClientId', () => {
+    it('should subscribe to the block tracker and update accounts if not already using the networkClientId', async () => {
+      jest.spyOn(blockTrackerFromHookStub, 'addListener').mockImplementation();
+      const updateAccountsWithNetworkClientIdSpy = jest
```

Comment:
[nit] method was since renamed
```suggestion
      const updateAccountsSpy = jest
```

Reply:
several instances of this below ☝️

Reply:
https://github.com/MetaMask/metamask-extension/pull/21868/commits/19ce6d1d25f207b838896ae2d0a1ef6df6f493f2